packages/react-admin/src/mui/list/FilterForm.js

```diff
@@ -88,60 +93,69 @@ export class FilterForm extends Component {
             className,
             resource,
             translate,
+            handleSubmit,
             ...rest
         } = this.props;
 
         return (
-            <div className={className} {...sanitizeRestProps(rest)}>
-                <CardContent className={classes.card}>
-                    {this.getShownFilters()
-                        .reverse()
-                        .map(filterElement => (
-                            <div
-                                key={filterElement.props.source}
-                                data-source={filterElement.props.source}
-                                className={classnames(
-                                    'filter-field',
-                                    classes.body
-                                )}
-                            >
-                                {filterElement.props.alwaysOn ? (
-                                    <div className={classes.spacer}>&nbsp;</div>
-                                ) : (
-                                    <IconButton
-                                        className="hide-filter"
-                                        onClick={this.handleHide}
-                                        data-key={filterElement.props.source}
-                                        tooltip={translate(
-                                            'ra.action.remove_filter'
-                                        )}
-                                    >
-                                        <ActionHide />
-                                    </IconButton>
-                                )}
-                                <div>
-                                    <Field
-                                        allowEmpty
-                                        {...filterElement.props}
-                                        name={filterElement.props.source}
-                                        component={filterElement.type}
-                                        resource={resource}
-                                        record={emptyRecord}
-                                    />
+            <form onSubmit={handleSubmit}>
```

Comment:
what's the interest of handling submission? Validation?

Reply:
Yes, I've tried to use: 
```
reduxForm({
  onChange: (values,dispatch,props)=>{
     props && props.valid && props.setFilters(values)
 } 
})
```
but `valid` is not updated/evaluated ontime @see: https://github.com/erikras/redux-form/issues/3754 

And either way, only `submit` allows the gui to nicely show the validation errors.